app/controllers/hub/state_file/automated_messages_controller.rb

```diff
@@ -1,26 +1,71 @@
 module Hub::StateFile
   class AutomatedMessagesController < Hub::StateFile::BaseController
+    include StateFile::SurveyLinksConcern
+    helper_method :email_message
+    helper_method :sms_body
 
     def index
-      @messages = messages_preview
+      Rails.application.eager_load!
+      @messages = StateFile::AutomatedMessage::BaseAutomatedMessage.descendants
+      @locales = [:en, :es]
+      @us_state = StateFile::StateInformationService.active_state_codes.include?(params[:us_state]) ? params[:us_state] : "az"
+      get_intake
     end
 
     private
 
-    def messages_preview
-      Rails.application.eager_load!
-      message_classes = StateFile::AutomatedMessage::BaseAutomatedMessage.descendants
-      message_classes.to_h do |klass|
-        replaced_body = klass.new.email_body.gsub('<<', '&lt;&lt;').gsub('>>', '&gt;&gt;')
-        email = StateFileNotificationEmail.new(to: "example@example.com",
-                                               body: replaced_body,
-                                               subject: klass.new.email_subject)
-        message = StateFile::NotificationMailer.user_message(notification_email: email)
-        ActionMailer::Base.preview_interceptors.each do |interceptor|
-          interceptor.previewing_email(message)
-        end
-        [klass, message]
+    def get_intake
+      return @intake if @intake.present?
+      state_class = "StateFile#{@us_state.titleize}Intake".constantize
+      if params[:intake_id].present?
+        @intake = state_class.find_by_id(params[:intake_id])
+        return @intake if @intake.present?
+        flash[:alert] = "Unknown Intake"
       end
+      @intake = state_class.new(
```

Comment:
so if there is not an intake id set then it'll use this, is there no id because its not actually creating it?

Reply:
Yup